dll.go

```diff
@@ -84,16 +86,80 @@ func gather(source string, asFile bool) ([]*report, error) {
 func main() {
 	if len(os.Args) < 2 {
 		fmt.Fprintln(os.Stderr, "no source files supplied")
+		os.Exit(1)
 	}
 
-	for _, source := range os.Args[1:] {
+	files := os.Args[1:]
+	fileCount := len(files)
+
+	reportsChannel := make(chan []*report, fileCount)
+	cpuCount := runtime.NumCPU()
+	filesPerCore := splitArrayIntoParts(files, cpuCount)
+
+	go func() {
```

Comment:
I think a simpler approach would be to just spawn a new goroutine for each file and add each slice of `*reports` to another slice guarded by a mutex. Then when the `WaitGroup` is `done()`, iterate through the slice of _all_ reports and dump that to stdout.

I don't know how much of an advantage the splitting gives us either as I usually let the Go runtime schedule goroutines for me. :)

Other than that I like the approach. I think if we pare it down to what I tried to describe, it'd be perfect.